File: lc_java/lcJava/array/minCosts3502.java
package array;



class minCosts3502 {
    public int[] minCosts(int[] cost){
        int n = cost.length;
        int[] ans = new int[n];
        ans[0] = cost[0];
        for(int i = 1;i<n;i++){
            ans[i] = Math.min(ans[i-1],cost[i]);
        }
        return ans;
    }
    public void main(){
        minCosts3502 sol = new minCosts3502();
        int[] cost = {5,3,1,2,4};
        int[] ans = sol.minCosts(cost);
        for(int i = 0;i<ans.length;i++){
            System.out.println(ans[i]);
        }
    }
}

File: lc_cpp/lcCpp/arraY/minCosts3502.cpp
//
// Created by 24276 on 2025/4/25.
//

#include <bits/stdc++.h>
#include <vector>

using namespace std;

class min_costs3502 {
public :
    vector<int> minCosts(vector<int>& cost) {
        int n = cost.size();
        vector<int> ans (n);
        ans[0] = cost[0];
        for (int i = 1;i < n;i++) {
            ans[i] = min(ans[i - 1],cost[i]);
        }
    return ans;
}
};

int main() {
    min_costs3502 sol;
    vector<int> cost = {5,3,4,1,3,2};
    auto res = sol.minCosts(cost);
    for (int ele : res) {
        cout << ele << " ";
    }
    return 0;
}
